Fix ugly text rendering in various places

harperstoeckle opened the pull request:
`Main.GUIBarsDrawInner` sets the sampler state to `PointClamp`, which ends up leaking into text rendering in a few places. This includes
- Mouse text in `IngameFancyUI`, including the bestiary.
- Mouse text in the options menu.
- Camera mode.

I believe this issue also affects vanilla.

Here are a few screenshots showing the difference before and after this patch:
| **Before** | **After** |
| --- | --- |
| ![image](https://github.com/user-attachments/assets/30bf2d5e-a9db-4383-a441-ddb0c0422faa) | ![image](https://github.com/user-attachments/assets/37742b63-7e55-47c3-b98a-d15f9120388e) |
| ![image](https://github.com/user-attachments/assets/4dcd59ed-bac4-43b3-9109-7fbf023b5f2d) | ![image](https://github.com/user-attachments/assets/31557f46-6bcd-4e40-b703-166d50719a60) |
| ![image](https://github.com/user-attachments/assets/e74b36b5-0a4a-455c-85db-5b7ba448ec1c) | ![image](https://github.com/user-attachments/assets/1c0c28f4-d2f3-4eab-9c08-e45f040c0b12) |

I opted to reset the sampler state at the end of `GUIBarsDrawInner`, but it could also probably be handled by setting it at the top of the affected functions (like `MouseTextInner`).

As far as I can tell, the parameters I give to `spriteBatch.Begin` are the same as the ones set prior to every call to `GUIBarsDrawInner`, but I'm not 100% sure. I think optimally, there would be some way to save and restore the sprite batch state, a bit like the proposal in #4072.

harperstoeckle committed:
Restore sampler state at the end of Main.GUIBarsDrawInner

JavidPack commented:
Note: This bug can only be replicated at non-100% UI zoom levels.

I've noticed other issues such as hover text and the cursor itself:

![NVIDIA_Share_2025-03-17_14-47-39](https://github.com/user-attachments/assets/a707a87e-0165-49a7-9f39-690667edadef)
![NVIDIA_Share_2025-03-17_14-47-48](https://github.com/user-attachments/assets/d99853f8-2ad6-4e64-bf70-9fd4bfe5a89b)
![NVIDIA_Share_2025-03-17_14-48-50](https://github.com/user-attachments/assets/8f22d119-f52e-4068-a471-14bb89af524e)
![NVIDIA_Share_2025-03-17_14-48-56](https://github.com/user-attachments/assets/100a3bb4-38c2-4dab-be60-2b4583a9faa5)
![NVIDIA_Share_2025-03-17_14-49-53](https://github.com/user-attachments/assets/343c4fe9-74eb-4742-a0b1-8f8058889c1a)
![NVIDIA_Share_2025-03-17_14-50-21](https://github.com/user-attachments/assets/adcb2873-27d4-409e-9722-1d59c64f48f0)

One other thing this PR changes as a result of the fix is the look of the Camera Mode buttons:

https://github.com/user-attachments/assets/546c4a11-e4de-45f5-b8d3-425c8e34ce31

harperstoeckle committed:
Use PointClamp when drawing camera mode buttons to match vanilla

487666123 commented:
"Personal opinion, the original pixel rendering mode was actually more comfortable."

harperstoeckle commented:
Yeah I just added a commit that should fix the camera buttons to match how they are drawn in vanilla. The text rendering, however, is a major issue, because it makes it almost impossible to read text in `IngameFancyUI`s with the UI zoomed out. I'm currently working on a mod that uses `IngameFancyUI`, and it's nigh unusable at low zoom levels, even though the normal inventory is perfectly fine.

JavidPack commented:
> Yeah I just added a commit that should fix the camera buttons to match how they are drawn in vanilla. The text rendering, however, is a major issue, because it makes it almost impossible to read text in `IngameFancyUI`s with the UI zoomed out. I'm currently working on a mod that uses `IngameFancyUI`, and it's nigh unusable at low zoom levels, even though the normal inventory is perfectly fine.

Are you saying with the fix it is bad? Or are you saying the fix will make it better?

harperstoeckle commented:
> > Yeah I just added a commit that should fix the camera buttons to match how they are drawn in vanilla. The text rendering, however, is a major issue, because it makes it almost impossible to read text in `IngameFancyUI`s with the UI zoomed out. I'm currently working on a mod that uses `IngameFancyUI`, and it's nigh unusable at low zoom levels, even though the normal inventory is perfectly fine.
> 
> Are you saying with the fix it is bad? Or are you saying the fix will make it better?

In its original form, this PR fixed the text rendering issue, but also changed how the camera buttons were being rendered, as you pointed out. I recently added a new commit that updates the camera buttons to look the same as they do in vanilla. I.e., in its current form, this PR fixes the text rendering problem without changing how the camera buttons look.

@487666123 said that they preferred the original pixel rendering, and I wasn't sure whether they were talking about the camera buttons or the text; In my reply, I was trying to respond to both. I agree that the camera buttons look better with the vanilla rendering, so the first sentence was me explaining that I had added the new commit. The rest of my reply was me explaining why I thought the text rendering changes introduced in this PR are necessary, in case they were saying that they preferred the original vanilla text rendering.

JavidPack committed:
Add comment so we know patch can be deleted if vanilla is fixed
